### ISSUE TITLE
Possible workaround for setting schema location

### ISSUE THREAD

**Issue by epakai** [johnfactotum/foliate#342]
Just going to outline how I got to this "solution":
1. I built a debian (buster / 10.4) package from master branch
2. there's some issue with directly doing debuild (something about appstream validation failing), but I'm ignoring that for now.
3. After install, com.github.johnfactotum.Foliate crashes with `(com.github.johnfactotum.Foliate:9635): GLib-GIO-ERROR **: 20:08:39.938: Settings schema 'com.github.johnfactotum.Foliate' does not contain a key named 'img-event-type'`
4. Looking at other gtk apps they build a schema dir like `path.join(install_prefix, 'share', 'glib-2.0', 'schemas')`
5. Other apps also use something like SettingsSchemaSource, but it appears it's not exposed in gjs yet.
6. I edited /usr/bin/com.github.johnfactotum.Foliate and added

> const GLib = imports.gi.GLib;
> GLib.setenv("GSETTINGS_SCHEMA_DIR", "/usr/share/glib-2.0/schemas", true);

Now it doesn't crash, and I don't need to set the environment variable as mentioned in the README.

Here's my changes to have this done in the meson build:
https://github.com/epakai/foliate/commit/149d919f04edbb3dde49841fa8a1841dbfae1ad6

**Comment by johnfactotum** [comment]
`GSettingsSchemaSource` is available in GJS under `imports.gi.Gio.SettingsSchemaSource`.

**Comment by epakai** [comment]
I looked into it more. It seems SettingsSchemaSource is unnecessary.

I feel like I'm throwing wild guesses here, but it looks like the schema I see in dconf-editor, and the schema in `/usr/share/glib-2.0/schemas/` are two different things. I previously used Foliate 1.4.x, but I'm guessing 2.x doesn't handle updating the schema for my current backend?

In any case setting the environment variable causes it to use the new schema, but without it I crash because the new keys aren't in the old schema.

dconf-editor doesn't let me delete things. I'm at a loss as to whether Foliate should be doing something to add the new keys in my current backend instead of just crashing out when it can't find them.

**Comment by johnfactotum** [comment]
`/usr/share/glib-2.0/schemas/` should be the default directory, so what you see in dconf-editor should be the result of `/usr/share/glib-2.0/schemas/gschemas.compiled`. If you've changed the schemas, you need to compile them with `glib-compile-schemas`, which is what the post-install script does: https://github.com/johnfactotum/foliate/blob/bfbd42497010f2f79bdaaade710ebfed064d2202/build-aux/meson/postinstall.py#L19

`GSettingsSchemaSource`/`GSETTINGS_SCHEMA_DIR` should only be needed when, for some reason, you can't or don't want to install the schema to `/usr/share/glib-2.0/schemas/`.

**Comment by epakai** [comment]
I poked around for a while. Finally noticed there was a second `gschemas.compiled` file in /usr/**_local_**/share/glib-2.0/schemas even though the schema there had been removed. It had priority even though the new schema was being compiled.

Removing this extra `gschemas.compiled` caused the newer schema to load. I don't think there's anything for Foliate to do in this case, just an unfortunate side effect of having had it installed in a different prefix at one time.
